### PR TITLE
Use Github as Preparer for the Scaffolder

### LINKED ISSUE #1370: Implement the Github Preparer for the Scaffolder
Because of the work completed in #1318, the scaffolder can now understand entities which are loaded with a file.

We now want the ability to do the same thing for Github remote repositories, and be able to clone the repository with an `ACCESS_TOKEN` and create the temporary directory with the contents of that repository and path filter.

### Feature Suggestion
- Implement a `preparer` for the Github source

### PR DESCRIPTION
**Pull request by benjdlambert**
- Adds the ability to use public Github Repositories as templates for the Scaffolder
- Doesn't have auth functionality for Private repos yet
- Depends on `mob/prepare-from-catalog`

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

Closes #1370

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/prepare/github.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    );
+
+    const checkoutOptions = new CheckoutOptions();
+    if (template.spec.path) {
+      checkoutOptions.paths = [templateDirectory];
+    }
+
+    await Clone.clone(repositoryCheckoutUrl, tempDir, {
+      checkoutOpts: checkoutOptions,
+      // TODO(blam): Maybe need some auth here?
```
Comment (Rugvip) [inline plugins/scaffolder-backend/src/scaffolder/prepare/github.ts:55]: Probably gonna be best to require that from the system as a whole and not the user, and we'll need to for the catalog. Not for this PR though ofc :grin:

File: plugins/scaffolder-backend/src/scaffolder/prepare/github.test.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      { checkoutOpts: {} },
+    );
+  });
+
+  it('return the temp directory with the path to the folder if it is specified', async () => {
+    const preparer = new GithubPreparer();
+    mockEntity.spec.path = './template/test/1/2/3';
+    const response = await preparer.prepare(mockEntity);
+
+    expect(response).toMatch(new RegExp(/\/template\/test\/1\/2\/3$/));
```
Comment (Rugvip) [inline plugins/scaffolder-backend/src/scaffolder/prepare/github.test.ts:83]: `expect(response).toMatch(/\/template\/test\/1\/2\/3$/)`? O_o

**Review by Rugvip — APPROVED** [review]
Good stuff 👌  😁

File: plugins/scaffolder-backend/src/scaffolder/prepare/github.test.ts (hunk 2 above)
Comment (benjdlambert) [reply plugins/scaffolder-backend/src/scaffolder/prepare/github.test.ts:83]: There is no `expect(string).toEndWith('asd')`

Comment (benjdlambert) [reply plugins/scaffolder-backend/src/scaffolder/prepare/github.test.ts:83]: I could do `expect(string.endsWith('asd')).toBe(true)`

**Pull request merged**